docs/discussion/Ratify v2 design scope.md

```diff
@@ -0,0 +1,154 @@
+
+**Proposed Solution**:
+1. Move the external plugins to a separate repo. This will reduce the dependencies of the project.
+2. Reorganize the codebase. Follow the best practices of Go project layout.
+3. Extract out the Ratify core and different user scenarios to different packages. As a first step, we could just create different folders for different scenarios. And in the future, we can consider moving them to different repos.
+
+**Priority**: Moving out external plugin implementation can be priority 2. Reorganizing the codebase can be priority 1 since it's the fundamental part of the project.
+
+### Deprecations
+#### CertificateStore
+**Current Limitation**:
+1. `CertificateStore` is a previous version of `KeyManagementProvider`. It's incompatible with multi-tenancy scenario and does not support the latest features of the `KeyManagementProvider`, such as key rotation.
+
+**Proposed Solution**:
+1. Deprecate `CertificateStore` and recommend customers to use `KeyManagementProvider` instead.
+
+**Priority**: 1
+
+#### Legacy Cosign Verifier
+**Current Limitation**:
+1. The legacy Cosign verifier does not support referencing multiple keys and KeyManagementProvider.
+2. The legacy Cosign verifier does not support trust policy for cosign verification.
+
+**Proposed Solution**:
+1. Deprecate the legacy Cosign verifier and recommend customers to use the new Cosign verifier.
+
+**Priority**: 1
+
+#### License Checker Verifier
+**Current Limitation**:
+1. The primitive implementation of the licensechecker verifier to support basic verification of license compliance (allowed license list).
+
+**Proposed Solution**:
+1. Deprecate the licensechecker verifier and recommend customers to use the new SBOM verifier.
+
+**Priority**: 1
+
+#### `Name` and `Type` fields in verifierReport/configuration
+**Current Limitation**:
+1. `Name` and `Type` fields in VerifierReport refer to the name and type of the verifier that generated the report. But we got feedback that they're ambiguous and can be misleading.
+2. `Name` and `Type` fields in plugin configuration are misused due to historical reasons. It makes the configuration hard to understand.
+
+**Proposed Solution**:
+1. Deprecate `Name` and `Type` fields in VerifierReport and recommend customers to use `VerifierName` and `VerifierType` instead.
+2. Correct the usage of `Name` and `Type` fields in plugin configuration.
+
+**Priority**: 1
+
+### Documentation Update
+Post or along with the v2 release, we should update the documentation to reflect the changes in the v2 design. The documentation update should include the following aspects:
+1. Announce deprecations and provide migration guides.
+2. Update the architecture and design documents.
+3. Update the user guide and CLI reference.
+4. Update the developer guide and plugin development guide.
+
+**Priority**: 3
```

Comment:
Hi @binbin-li , i believe doc has great value for customers to onboard to Ratify V2. I would suggest doc updates are also pri1

Reply:
yeah, in terms of recent feedback from community, doc is really important to guide new users to Ratify.